src/main/java/com/microsoft/dhalion/CompositeMetricsProvider.java

```diff
@@ -0,0 +1,109 @@
+package com.microsoft.dhalion;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.microsoft.dhalion.api.MetricsProvider;
+import com.microsoft.dhalion.conf.Config;
+import com.microsoft.dhalion.conf.Key;
+import com.microsoft.dhalion.core.Measurement;
+
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class CompositeMetricsProvider implements MetricsProvider {
+
+  private List<MetricsProvider> metricsProviders;
+  private Map<String, MetricsProvider> metricsProviderMap;
+  protected Config sysConfig;
+
+  @Inject
+  public CompositeMetricsProvider(Config sysConf) throws ClassNotFoundException {
+    sysConfig = sysConf;
+    metricsProviders = new ArrayList<>();
+    metricsProviderMap = new HashMap<>();
+    instantiateMetricsProviders();
+  }
+
+  @VisibleForTesting
+  protected void instantiateMetricsProviders() throws ClassNotFoundException {
+    Injector injector = Guice.createInjector(new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(Config.class).toInstance(sysConfig);
+      }
+    });
+    String metricsProviderClasses = (String) sysConfig.get(Key.METRICS_PROVIDER_CLASS.value());
```

Comment:
`metricsProviderClasses` could be null?

Reply:
This config was previously used for instantiating metrics provider class. I assume, that it can never be null.